langchain/src/document_loaders/fs/unstructured.ts

```diff
@@ -95,25 +115,26 @@ export class UnstructuredLoader extends BaseDocumentLoader {
 
 export class UnstructuredDirectoryLoader extends DirectoryLoader {
   constructor(
-    public webPath: string,
     public directoryPath: string,
+    public webPath: string = "https://api.unstructured.io/general/v0/general",
+    public options: UnstructuredOptions = {},
     public recursive: boolean = true,
     public unknown: UnknownHandling = UnknownHandling.Warn
   ) {
     const loaders = {
-      ".txt": (p: string) => new UnstructuredLoader(webPath, p),
-      ".text": (p: string) => new UnstructuredLoader(webPath, p),
-      ".pdf": (p: string) => new UnstructuredLoader(webPath, p),
-      ".docx": (p: string) => new UnstructuredLoader(webPath, p),
-      ".doc": (p: string) => new UnstructuredLoader(webPath, p),
-      ".jpg": (p: string) => new UnstructuredLoader(webPath, p),
-      ".jpeg": (p: string) => new UnstructuredLoader(webPath, p),
-      ".eml": (p: string) => new UnstructuredLoader(webPath, p),
-      ".html": (p: string) => new UnstructuredLoader(webPath, p),
-      ".md": (p: string) => new UnstructuredLoader(webPath, p),
-      ".pptx": (p: string) => new UnstructuredLoader(webPath, p),
-      ".ppt": (p: string) => new UnstructuredLoader(webPath, p),
-      ".msg": (p: string) => new UnstructuredLoader(webPath, p),
+      ".txt": (p: string) => new UnstructuredLoader(webPath, p, options),
```

Comment:
To cut down on repetition, can we do something like:
```typescript
const FILETYPES = [
  ".txt",
  ".text",
  ".pdf",
  ...
];
const loaders = FILETYPES.reduce((loadersObject, filetype) => {
  loadersObject[filetype] = (p: string) => new UnstructuredLoader(webPath, p, options);
  return loadersObject;
}, {});
```

Reply:
Thanks, just added that in there.